### PR TITLE
fix(retrospect): gate-7 scans exit-0 content errors

### PR DESCRIPTION
**Pull request by devseunggwan**
## 배경

Stage-2 friction 열거와 Gate-7 receipt는 `grep -c '"is_error":true'`에 앵커링됩니다.
harness는 **도구 호출 자체**가 실패할 때만 `is_error:true`를 세팅하므로, exit-0
`tool_result` **content**에 박힌 에러(비-0 종료 Bash의 "Exit code 1", 래퍼 exit-0
"Usage:"·"FATAL", 브라우저/MCP의 "js_error:")는 구조적으로 누락됩니다.

## 변경

`hooks/completion-verify/retrospect-mix-check/`:

1. **content-error-signal 스캔 + per-signal disposition enum** (구조 검사)
   - receipt에 `content_error_count: N` 필드 요구. 부재 시 block.
   - `> 0`이면 `retrospect:content_error_enum` 블록(promote|note|dismiss disposition 행)을
     요구 — count만으로는 통과 불가 (#664 is_error_enum 미러).
   - 캘리브레이션 regex(에러 **구문** 매칭, bare word "error" 아님 → `*_error`·`"is_error":false`
     false positive 회피): `Exit code [1-9]|command terminated|js_error|FATAL|No such file|denied|BLOCKED|Usage: `

2. **floor 재도출** (#671 value-check 확장) — declared 0 laundering 차단
   - hook이 transcript에서 직접 재도출하되 `"type":"tool_result"` 라인에 **한정**
     (assistant 산문·Stage-3 리포트는 그 라인이 아니라 배제 → over-block 방지).
   - declared `content_error_count: 0`인데 live tool_result 스캔이 tolerance 초과면 block.
   - exact equality 아닌 floor: disposition으로 dismiss는 가능하나 신호가 있는데 0 주장은 불가.

## 검증

- `tests/hooks/completion-verify/test_retrospect_mix_check.sh` → **84 cases / 11 fixtures passed**
  - 신규: CE1~CE7(구조·캘리브레이션) + CE_FLOOR1~3(floor: declared-0+live errors→block,
    prose-only→pass(scoped out), 1신호→pass(tolerance))
- `./scripts/run-tests.sh` → **ALL TESTS PASSED** (345 pytest 포함)

Caller chain verified: N/A -- hook invoked by Claude Code runtime (hooks/manifest.json)
Pre-commit verified: ./scripts/run-tests.sh -> ALL TESTS PASSED (84 hook cases, 11 fixtures, 345 pytest)

Closes #670

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced transcript error validation to detect and properly track embedded tool errors missed by previous checks, with stricter enforcement of error declarations and corresponding documentation requirements.

* **Tests**
  * Added comprehensive test coverage for content error detection and validation scenarios, including edge cases and tolerance threshold validation.

### CODE REVIEW

File: hooks/completion-verify/retrospect-mix-check/impl.sh [hunk 1]
```diff
@@ -410,6 +412,42 @@ if grep -Eq '(^|[^\\])"isCompactSummary"[[:space:]]*:[[:space:]]*true' "$TRANSCR
...
+      # content_error_count sits on the pipe-delimited count line (not its own line),
+      # so do NOT anchor with '^'. Use a word-boundary-style guard: require the field
+      # to be preceded by a non-digit non-letter char (pipe/space) or start of string
+      # to resist injection via enum rows that could quote 'content_error_count: 99'.
+      # Pick the first match (head -1) to be consistent with ie_count extraction.
+      ce_count=$(printf '%s\n' "$RECEIPT_BLOCK" \
+        | grep -oE '[^a-zA-Z0-9_]content_error_count:[[:space:]]*[0-9]+|^[[:space:]]*content_error_count:[[:space:]]*[0-9]+' \
+        | grep -oE 'content_error_count:[[:space:]]*[0-9]+' | grep -oE '[0-9]+' | head -1)
+      if [ -z "$ce_count" ]; then
+        GATE7_VIOLATION="post-compaction transcript_receipt fence carries no parseable 'content_error_count: N' line inside the fence — run grep -cE 'Exit code [1-9]|command terminated|js_error|FATAL|No such file|denied|BLOCKED|Usage: ' against the transcript content and paste the REAL count in the receipt before Stage 3"
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply hooks/completion-verify/retrospect-mix-check/impl.sh]: Fixed in 8aaadc0 — the guidance command is now scoped to `"type":"tool_result"` lines, matching the floor re-derivation and the spec, so it no longer inflates the count from non-tool_result lines.

File: hooks/completion-verify/retrospect-mix-check/impl.sh [hunk 2]
```diff
@@ -437,6 +477,13 @@ if grep -Eq '(^|[^\\])"isCompactSummary"[[:space:]]*:[[:space:]]*true' "$TRANSCR
       # Re-derive user_turn_count (role:user covers both literal JSON forms).
       live_ut_count=$(grep -c '"role":"user"' "$TRANSCRIPT_PATH" 2>/dev/null || true)
       live_ut_count=${live_ut_count:-0}
+      # Re-derive content_error_count [issue #670], scoped to tool_result-bearing
+      # lines so the calibrated regex matches errors embedded in exit-0 tool_result
+      # CONTENT, not the agent's own analysis prose or the Stage-3 report (those
+      # live in assistant/text lines, never carry "type":"tool_result").
+      live_ce_count=$(grep '"type":"tool_result"' "$TRANSCRIPT_PATH" 2>/dev/null \
+        | grep -cE 'Exit code [1-9]|command terminated|js_error|FATAL|No such file|denied|BLOCKED|Usage: ' 2>/dev/null || true)
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply hooks/completion-verify/retrospect-mix-check/impl.sh]: Fixed in 8aaadc0 — regex broadened to `Usage:([[:space:]]|$)` across all occurrences (floor grep, guidance messages, comment, tests). CE7 calibration now includes a bare `Usage:` line and expects 9 matches.
